### PR TITLE
steps/assets: split to a base module and a platform-specific one, add libvirt

### PR DESCRIPTION
**Pull request by squeed**
This does the following:
* Move most of asset generation to assets/base
* Consumes this "base" module in assets/aws
* Moves aws-specific variables to variables-aws.tf
* Adds the beginning of libvirt support

### CODE REVIEW

**Comment by enxebre** [comment]
ok to test

File: steps/variables.tf [hunk 1]
```diff
@@ -1,364 +1,3 @@
...
-(optional) List of ELBs to attach all worker instances to.
-This is useful for exposing NodePort services via load-balancers managed separately from the cluster.
-
-Example:
- * `["ingress-nginx"]`
-EOF
-}
-
 // This variable is generated by tectonic internally. Do not modify
 variable "tectonic_cluster_id" {
```
Comment (enxebre) [inline steps/variables.tf]: couldn't we move this to config.tf and get rid of the variables.tf file?

**Comment by enxebre** [comment]
Thanks, Looking great to me!
Aside from this PR just thinking out loud, I think we do need to focus on moving this assets generation into the TNC and the cli

File: steps/assets/base/variables2.tf [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# is where we can define variables that the steps can pass directly
+variable "etcd_count" {
+  type = "string"
+}
+
+variable "cloud_provider" {
+  type = "string"
+}
+
+variable "ingress_kind" {
```
[the remark this answers is not in the document]
Comment (squeed) [reply steps/assets/base/variables2.tf]: Oh, that's definitely a problem; ingress needs to be HostPort for bare-metal situations.

**Comment by squeed** [comment]
rebased and updated after feedback

**Comment by squeed** [comment]
@enxebre you're right. it will be a lot easter to iterate on tnc + bootstrapping once libvirt is working :-)

**Comment by yifan-gu** [comment]
retest this please

**Comment by yifan-gu** [comment]
retest this please

**Comment by squeed** [comment]
retest this please

**Comment by squeed** [comment]
duh, the problem is that I renamed the platform.

**Comment by enxebre** [comment]
@squeed as far as I can see test are failing because the addon operator is failing to come up because there's no perms to download the image anymore, nothing to do with this PR
```Warning  Failed                 26m (x4 over 27m)   kubelet, ip-10-0-32-216.eu-west-1.compute.internal  Failed to pull image "quay.io/coreos/kubernetes-addon-operator:beryllium-m1": rpc error: code = Unknown desc = Error: Status 403 trying to pull repository coreos/kubernetes-addon-operator: "{\"error\": \"Permission Denied```

**Comment by enxebre** [comment]
retest this please

**Pull request closed without merging**